Assets/Packages/com.studio23.ss2.authsystem.steam/Runtime/Core/SteamProvider.cs

```diff
@@ -0,0 +1,31 @@
+using Steamworks;
+using Studio23.SS2.AuthSystem.Data;
+using UnityEngine;
+
+namespace Studio23.SS2.AuthSystem.Steam.Core
+{
+    public class SteamProvider : ProviderBase
+    {
+        public uint AppId;
+
+        public void SetAppId(uint appId)
+        {
+            AppId = appId;
+        }
+
+        /// <summary>
+        /// Call this method to enable steam authentication per user
+        /// </summary>
+        public override void Authenticate()
+        {
+            if (SteamClient.RestartAppIfNecessary(AppId))
+            {
+                SteamClient.Shutdown();
+                Application.Quit();
+            }
+
+            SteamClient.Init(AppId);
```

Comment:
Do we get any sort of call back from the `Init` function? If so, there should be an exception/error handling when the authentication fails for some reason. (ie. network issues.)

Reply:
There's no need to enable callback for exception handling. `Init()` method as it already handles exception on itself whenever it tries to initialize using appId.